src/common/LoadingBanner.js

```diff
@@ -2,75 +2,56 @@
 
 import React, { PureComponent } from 'react';
 import { View } from 'react-native';
+import Bar from 'react-native-progress/Bar';
 
 import type { Dispatch } from '../types';
 import { connect } from '../react-redux';
 import { getLoading } from '../selectors';
-import { Label, LoadingIndicator } from '.';
 import type { ThemeData } from '../styles';
-import { ThemeContext, createStyleSheet } from '../styles';
-
-const key = 'LoadingBanner';
-
-const styles = createStyleSheet({
-  block: {
-    flexDirection: 'row',
-    justifyContent: 'center',
-    alignItems: 'center',
-    backgroundColor: 'hsl(6, 98%, 57%)',
-  },
-  none: { display: 'none' },
-});
+import { ThemeContext, BRAND_COLOR, NAVBAR_SIZE } from '../styles';
 
 type SelectorProps = $ReadOnly<{|
   loading: boolean,
 |}>;
 
 type Props = $ReadOnly<{|
-  spinnerColor?: 'black' | 'white' | 'default',
-  textColor?: string,
-  backgroundColor?: string,
+  color: string,
+  unfilledColor: string | null,
 
   dispatch: Dispatch,
   ...SelectorProps,
 |}>;
 
 /**
- * Display a notice that the app is connecting to the server, when appropriate.
+ * Display a indeterminate loading-bar while the app is connecting to the server, when appropriate.
  */
 class LoadingBanner extends PureComponent<Props> {
   static contextType = ThemeContext;
   context: ThemeData;
 
+  static defaultProps = {
+    color: BRAND_COLOR,
+    unfilledColor: null,
+  };
+
   render() {
-    if (!this.props.loading) {
-      return <View key={key} style={styles.none} />;
-    }
-    const {
-      spinnerColor = 'default',
-      textColor = this.context.color,
-      backgroundColor = this.context.backgroundColor,
-    } = this.props;
-    const style = {
-      width: '100%',
-      flexDirection: 'row',
-      justifyContent: 'center',
-      alignItems: 'center',
-      backgroundColor,
-    };
+    const { loading, color, unfilledColor } = this.props;
+    const barHeight = 3;
     return (
-      <View key={key} style={style}>
-        <View>
-          <LoadingIndicator size={14} color={spinnerColor} />
-        </View>
-        <Label
-          style={{
-            fontSize: 14,
-            margin: 2,
-            color: textColor,
-          }}
-          text="Connecting..."
-        />
+      <View style={{ position: 'absolute', left: 0, right: 0, top: NAVBAR_SIZE - barHeight }}>
```

Comment:
Can this say something like `bottom: 0`, instead of `top:` with arithmetic? That would help keep it more robust to being used in places that have a height different from `NAVBAR_SIZE`.

Reply:
yes right but, look into the file `main/StreamTabsScreen.js ` something i already said above and now if `bottom:0` will be used then It will definetly be at the bottom of screen, same as you said 

> It looks like this will put the loading bar/banner at the bottom of this view

In first push I did same as you are saying but because of the StreamTabsScreen.js I changed to the current we have, I assume you maybe think about put loadingBar in `StreamListCard` and `SubscriptionsCard` for that we need `top=0` and the reason I didn't implement like that is we will have two bars(one loadingBar below and other (in half screen) TabBar-Indicator above and that will look weired. To make them concurrent I used loading-bar in parent component of `StreamListCard` and `SubscriptionsCard` that is `main/StreamTabsScreen.js `

>That would help keep it more robust to being used in places that have a height different from NAVBAR_SIZE.

we can have props to accomplish that

Reply:
Yeah, now that I see what's going on with the `position: 'absolute'`, it sounds like this is already very fragile. This is placing itself at a particular spot on the screen, which is dependent on the rest of what's on the screen around it looking a very particular way -- if there isn't some kind of bar at the top of height NAVBAR_SIZE, then this is going to look completely nonsense.

Maybe we end up doing that, but if so it's necessary that when you're looking at the code of any of the places where this is invoked, it's easy to tell that there's this unusual extra constraint that it makes on the layout of the rest of that screen. Otherwise as future changes happen to the app, we will end up with that kind of nonsense layout somewhere, with a loading indicator just plopping itself across the middle of some content in some other widget.

Reply:
Here's a way to address this that I think will give in the code a much better handle on how this layout works:

* Take the wrapper `View` with the `position: absolute`, and push it out to this component's callers. That way, the `position: absolute` is right there next to the source code of each surrounding `View`, the one that's being used to lay it out in each different context where we use it.
  * You can say something like `export const loadingBarHeight = 3` so that the callers can import that value, for those callers that need it.
  * This is enough to solve the really bad problem with the way this current version is using `position: 'absolute'`. We'll at least be able to see it's there, next to the code that controls each of the layouts it interacts with.
* Then, for I think probably all the places where we use this other than `StreamTabsScreen`, we should be able to push the `LoadingBar` inside the navbar, or whatever else it is that's expected to have height NAVBAR_SIZE.
  * That way, we avoid having a dependency on particular things *actually* being height NAVBAR_SIZE.
    * For example: in `HomeScreen`, is that row of icons at the top exactly NAVBAR_SIZE tall? Maybe it is, but it's sure not easy to tell that from its source code. Even if it is now, it might not be in the future, and it'd be hard to know when we're breaking that.
  * This will typically also mean that we can replace `top:` and arithmetic with `bottom: 0`.
* Then, for `StreamTabsScreen` and also if there are any other places where we end up still having a long-range dependency on the assumption that something is NAVBAR_SIZE tall, we make sure to add comments pointing out that dependency.

Reply:
>kind of nonsense layout somewhere, with a loading indicator just plopping itself across the middle of some content in some other widget

:D :D
well , I changed loadingBarHeight to LOADING_BAR_THICKNESS, I think its more suitable